packages/contracts-bedrock/tasks/genesis.ts

```diff
@@ -0,0 +1,268 @@
+    for (let i = 0; i <= 0xffff; i++) {
+      const num = ethers.utils.hexZeroPad('0x' + i.toString(16), 2)
+      const addr = ethers.utils.getAddress(
+        ethers.utils.hexConcat([prefix, num])
+      )
+
+      // There is no proxy at OVM_ETH or the GovernanceToken
+      if (
+        addr === ethers.utils.getAddress(predeploys.OVM_ETH) ||
+        addr === ethers.utils.getAddress(predeploys.GovernanceToken)
+      ) {
+        continue
+      }
+
+      alloc[addr] = {
+        nonce: '0x0',
+        balance: '0x0',
+        code: proxy.deployedBytecode,
+        storage: {
+          [adminSlot]: hre.deployConfig.proxyAdmin,
+        },
+      }
+
+      if (predeployAddrs.has(ethers.utils.getAddress(addr))) {
+        alloc[addr].storage[implementationSlot] = toCodeAddr(addr)
+      }
+    }
+
+    // Set the GovernanceToken in the state
+    // Cannot easily set storage due to no easy access to compiler
+    // output
+    const governanceToken = await hre.deployments.getArtifact('GovernanceToken')
+    alloc[predeploys.GovernanceToken] = {
+      nonce: '0x0',
+      balance: '0x0',
+      code: governanceToken.deployedBytecode,
+    }
+
+    // Give each predeploy a single wei
+    for (let i = 0; i <= 0xff; i++) {
+      const buf = Buffer.alloc(2)
+      buf.writeUInt16BE(i, 0)
+      const addr = ethers.utils.hexConcat([
+        '0x000000000000000000000000000000000000',
+        ethers.utils.hexZeroPad(buf, 2),
+      ])
+      alloc[addr] = {
+        balance: '0x1',
+      }
+    }
+
+    if (hre.deployConfig.fundDevAccounts) {
+      const accounts = [
+        '0xde3829a23df1479438622a08a116e8eb3f620bb5',
+        '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266',
+        '0x70997970C51812dc3A010C7d01b50e0d17dc79C8',
+      ]
+
+      const signers = await hre.ethers.getSigners()
+      for (const signer of signers) {
+        accounts.push(signer.address)
+      }
+
+      for (const account of accounts) {
+        alloc[account] = {
+          balance:
+            '0x200000000000000000000000000000000000000000000000000000000000000',
+        }
+      }
+    }
+
+    // Set the predeploys in the state
+    for (const [name, addr] of Object.entries(predeploys)) {
+      if (name === 'GovernanceToken') {
+        continue
+      }
+      const artifact = await hre.artifacts.readArtifact(name)
+      assertEvenLength(artifact.deployedBytecode)
+
+      const allocAddr = name === 'OVM_ETH' ? addr : toCodeAddr(addr)
+      assertEvenLength(allocAddr)
+
+      alloc[allocAddr] = {
+        nonce: '0x00',
+        balance: '0x00',
+        code: artifact.deployedBytecode,
+        storage: {},
+      }
+
+      const storageLayout = await getStorageLayout(hre, name)
+      const slots = computeStorageSlots(storageLayout, variables[name])
+
+      for (const slot of slots) {
+        alloc[allocAddr].storage[slot.key] = slot.val
+      }
+    }
+
+    const genesis: Genesis | any = {
```

Comment:
Modern `geth` seems to like different types than the geth we are using for the current system, this is why the type is extended to `any`. Will require a breaking change for the type, so a major version of core-utils or packages that use this type need to be pinned but that isn't useful for outside devs